### PR TITLE
Implement direct color and palette modes on the controller

### PR DESCRIPTION
**Pull request by ademuri**

### CODE REVIEW

File: lib/effect/FireflyEffect.cpp [hunk 1]
```diff
@@ -12,6 +12,15 @@ FireflyEffect::FireflyEffect() : Effect() {
 CRGB FireflyEffect::GetRGB(uint8_t led_index, uint32_t time_ms,
                            const StripDescription &strip,
                            RadioPacket *setEffectPacket) {
+  if (strip.FlagEnabled(Controller)) {
+    // For the controller, blink the lights mostly in sync
+    if (led_index % 2 == 1) {
```
Comment (wmmiii) [inline lib/effect/FireflyEffect.cpp:17]: Oh yeah, this confused me. Maybe give every-other light an 1/2 cycle time offset instead?

File: src/devices/controller/leds.cc [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#include "leds.h"
+
+std::array<CRGB, kLedCount> leds;
+
+void SetMainLed(uint8_t led_index, CRGB rgb) {
```
Comment (wmmiii) [inline src/devices/controller/leds.cc:6]: What does "Main" mean here?

File: src/devices/controller/controller.cpp [hunk 3]
```diff
@@ -1,35 +1,53 @@
...
-const StripDescription kRowStrip = StripDescription(/*led_count=*/12, {Bright});
+const StripDescription kRowStrip =
+    StripDescription(/*led_count=*/12, {Bright, Controller});
 const DeviceDescription kRowDescription(2000, {kRowStrip});
+const StripDescription kPaletteStrip =
+    StripDescription(/*led_count=*/5, {Bright, Controller});
 
 constexpr uint8_t kSetEffectDelay = 60;
 
+static constexpr uint16_t kMode1Threshold = 346 / 2;
```
Comment (wmmiii) [inline src/devices/controller/controller.cpp:51]: I'm confused by these magic numbers, what do they represent? Trial and error?

File: src/devices/controller/controller.cpp [hunk 4]
```diff
@@ -109,7 +146,196 @@ void loop() {
...
+    SetRightButtonLeds(kButtonActiveBrightness, kButtonActiveBrightness,
+                       kButtonPressedBrightness);
+  } else {
+    SetRightButtonLeds(kButtonActiveBrightness, kButtonActiveBrightness,
+                       kButtonActiveBrightness);
+  }
+
+  SetBottomButtonLeds(0, 0, 0);
 
+  if (pressed) {
```
Comment (wmmiii) [inline src/devices/controller/controller.cpp:216]: Is this debounced? Does the call below block?

File: src/devices/controller/controller.cpp [hunk 5]
```diff
@@ -109,7 +146,196 @@ void loop() {
...
+      FastLED.show();
+    }
+  }
+  FastLED.clear(/*writeData=*/true);
+}
+
+void loop() {
+  state_machine.Tick();
+
+  left_buttons.Tick();
```
Comment (wmmiii) [inline src/devices/controller/controller.cpp:312]: Nit: (for follow up) I think I like "flush" better than "Tick" if all these are doing is flushing the state to the LED strip

File: src/devices/controller/controller.cpp (hunk 4 above)
Comment (ademuri) [reply src/devices/controller/controller.cpp:216]: Yes, `AnalogButton` is debounced (although v1.1 will eliminate it). `state_machine.SetEffect(&control_packet)` does block briefly (about 1ms).

File: src/devices/controller/controller.cpp (hunk 5 above)
Comment (ademuri) [reply src/devices/controller/controller.cpp:312]: Acknowledged, will update in a follow-up.

File: lib/effect/FireflyEffect.cpp (hunk 1 above)
Comment (ademuri) [reply lib/effect/FireflyEffect.cpp:17]: I think this that this makes this effect more visually distinctive on the controller - I'd like to leave as-is for now, and if it's not clear when you've got a controller in-hand, we can update it.

File: src/devices/controller/leds.cc (hunk 2 above)
Comment (ademuri) [reply src/devices/controller/leds.cc:6]: Added a comment - it's for the 3 x 12 grid of LEDs.

File: src/devices/controller/controller.cpp (hunk 3 above)
Comment (ademuri) [reply src/devices/controller/controller.cpp:51]: Added a comment - it comes from a calculator.l

File: src/devices/controller/controller.cpp [hunk 6]
```diff
@@ -1,35 +1,57 @@
...
 #include "analog-button.h"
-#include "arduino-timer.h"
 #include "arduino/RadioHeadRadio.hpp"
+#include "buttons.h"
 #include "generic/NetworkManager.hpp"
 #include "generic/RadioStateMachine.hpp"
+#include "leds.h"
+
+    // Which mode of control the device is in
+    enum class ControllerMode {
```
Comment (wmmiii) [inline src/devices/controller/controller.cpp]: Why is this indented like this?

**Review by wmmiii — APPROVED** [review]

File: src/devices/controller/controller.cpp (hunk 6 above)
Comment (ademuri) [reply src/devices/controller/controller.cpp]: Oops, there was a stray 'h'. Removed it, and fixed this formatting.

**Pull request merged**